packages/react-components/src/shared/src/resolveChildren.ts

```diff
@@ -0,0 +1,19 @@
+import { ReactNode, ReactElement, Fragment } from "react";
+import { isFunction, isNil } from "lodash";
+
+// Support first level fragment: https://github.com/facebook/react/issues/11859.
+export function resolveFragment(children: ReactNode) {
+    if (!isNil(children) && (children as ReactElement).type === Fragment) {
+        return (children as ReactElement).props.children;
+    }
+
+    return children;
+}
+
+export function resolveChildren<T extends any[]>(children: ReactNode | ((args: T[]) => ReactNode), renderProps?: T): ReactNode {
```

Comment:
I think `renderProps` should be a `Record<string, any>`. It's basically an object literal.

Not sure about the `args: T[]`, There shouldn't be any case where the function can receive an array of renderProps

Reply:
In this case, T means i don't really care what the type is. As long as the function passed as the children parameter accept whatever is passed as the 2nd argument.